test/dark-gravity-wave-test.js

```diff
@@ -1,126 +1,286 @@
+        });
+
+        it('should throw error for signed int', () => {
+            expect(() => {
+                dgw.isValidTarget(-1, blocks);
+            }).to.throw('byte array longer than desired length');
+        });
+
+        it('should throw error when compact is undefined', () => {
+            expect(() => {
+                dgw.isValidTarget(undefined, blocks);
+            }).to.throw('Compact of type undefined not supported');
+        });
+
+        it('should throw error when compact is boolean', () => {
+            expect(() => {
+                dgw.isValidTarget(true, blocks);
+            }).to.throw('Compact of type boolean not supported');
+        });
+
+        it('height values ignored in calculation in getTarget function', () => {
+            //TODO is it true?
+            //and we can pass any blocks height: invalid, duplicates, not ordered...?
+            const highTargetBits = 0x1b0777d4;
+            const blocks2 = JSON.parse(JSON.stringify(blocks));
+            blocks2.forEach(function(value){
+                delete value["height"];
+            });
+            const result = dgw.isValidTarget(highTargetBits, blocks2);
+            expect(result).to.equal(true);
+        });
+
+        it('should throw error when compact is not defined in blocks', () => {
+            const highTargetBits = 0x1b0777d4;
+            const blocks2 = JSON.parse(JSON.stringify(blocks));
+            blocks2.forEach(function(value){
+                delete value["target"];
+            });
+            expect(() => {
+                dgw.isValidTarget(highTargetBits, blocks2);
+            }).to.throw('Compact of type undefined not supported');
+        });
+
+        it('should be invalid when timestamp values not specified', () => {
+            const highTargetBits = 0x1b0777d4;
+            const blocks2 = JSON.parse(JSON.stringify(blocks));
+            blocks2.forEach(function(value){
+                delete value["timestamp"];
+            });
+            const result = dgw.isValidTarget(highTargetBits, blocks2);
+            expect(result).to.equal(false);
+       });
+
+        it('should be invalid for blockTime=0', () => {
+            const highTargetBits = 0x1b0777d4; // 453474260
+            const result = dgw.isValidTarget(highTargetBits, blocks,  blockTime = 0);
+            expect(result).to.equal(false);
+        });
+
+        it('should be invalid for blockTime negative', () => {//TODO We should not allow negative blockTime
+            const highTargetBits = 0x1b0777d4; // 453474260
+            const result = dgw.isValidTarget(highTargetBits, blocks,  blockTime = -1);
+            expect(result).to.equal(true);//TODO change on true!
+        });
+
+        it('should be invalid when timestamp for some blocks grater then next one', () => {//TODO We should check if timestamp is grater then previous
+            const highTargetBits = 0x1b0777d4;
```

Comment:
more or less similar reasoning as in line 209.  however we might consider notifying the dev with an exception if things  like this is picked up.  Then again for `timestamp` in specific I am guessing that there is edge cases where timestamps can be out of order and still valid - Im not really sure.  Leaning towards not checking these fields.